sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala

```diff
@@ -112,6 +113,11 @@ class SQLContext private[sql](
    */
   def newSession(): SQLContext = new SQLContext(sharedState, isRootContext = false)
 
+  /** Pre-Load built-in User Defined Types */
+  if (isRootContext) {
```

Comment:
As a warning log will be shown when a userclass is registered already, removing this condition check will result too many log message in jenkins. I think I should add this back.